packages/m2m-gateway/src/services/agreementService.ts

```diff
@@ -157,6 +160,63 @@ export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
 
       const polledResource = await pollAgreement(response, headers);
 
+      return toM2MGatewayApiAgreement(polledResource.data);
+    },
+    async suspendAgreement(
+      agreementId: AgreementId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Agreement> {
+      logger.info(`Suspending agreement with id ${agreementId}`);
+
+      const response = await clients.agreementProcessClient.suspendAgreement(
+        undefined,
+        {
+          params: { agreementId },
+          headers,
+        }
+      );
+
+      const polledResource = await pollAgreement(response, headers);
+
+      return toM2MGatewayApiAgreement(polledResource.data);
+    },
+    async unsuspendAgreement(
+      agreementId: AgreementId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Agreement> {
+      logger.info(`Unsuspending agreement with id ${agreementId}`);
+
+      const agreement = await retrieveAgreementById(headers, agreementId);
+
+      assertAgreementIsSuspended(agreement.data);
```

Comment:
[same considerations](https://github.com/pagopa/interop-be-monorepo/pull/1911#discussion_r2101955371)

Reply:
Same: https://github.com/pagopa/interop-be-monorepo/pull/1911#discussion_r2102503528